### PR TITLE
Deep Q-Learning poker agent with softmax policy and curriculum learning

### PR DESCRIPTION
**Pull request by atorch**
## Overview
Implements a Deep Q-Learning agent for 3-player Texas Hold'em poker using SARSA with function approximation. After initial implementation revealed pathological behavior (agent folding premium hands), we identified the root cause and implemented a solution leveraging mixed-strategies.

## Problem Discovery
Initial implementation with deterministic argmax policy collapsed during self-play:
  - Agent learned to fold pocket aces and kings
  - Fold rate increased to 78% by episode 290
  - Episodes became extremely long (300+ deals)
  - **Root cause**: Argmax policy + self-play creates "folding is safe" feedback loop

## Solution
Three key improvements:

**1. Softmax Policy**
  - Replace deterministic argmax with probabilistic softmax sampling
  - Enables mixed strategies essential for Nash equilibrium in poker
  - Temperature parameter (default=1.0) controls exploration/exploitation
  - Comprehensive test coverage (6 new unit tests)

**2. Expanded State Space (16 → 19 inputs)**
  - Add pot size: enables pot odds calculations
  - Add opponent wealths: enables stack-size strategic play
  - Add opponent fold status: 1v1 vs 1v2 requires different strategies

**3. Curriculum Learning**
  - Phase 1 (episodes 0-100): Beat random agents
  - Phase 2 (episodes 100-200): Mixed opponents (50% random, 50% learned)
  - Phase 3 (episodes 200-400): Self-play
  - Prevents collapse into degenerate equilibria

## Results (400 episodes)
  ✅ **68% win rate** (vs 33% fair share in 3-player game)
  ✅ **All sanity checks pass:**
    - Premium hands (AA, KK, AK) prefer betting over folding
    - Q(check) > Q(fold) when checking is free
    - Q-values properly ordered by hand strength

  **Action distribution:** 31% fold, 62% positive bets (reasonable poker play)
  **Convergence:** Fast (avg 18 deals/episode, median 15)

## Key Learnings
  - **Softmax is essential for poker**: Deterministic policies lead to exploitable/degenerate strategies
  - **State representation matters**: Pot odds and opponent information enable strategic play
  - **Curriculum learning prevents collapse**: Random → Mixed → Self-play is more stable than pure self-play
  - **Self-play asymmetry**: Win rate stays above 33% because learning agent updates during episodes while opponents are frozen (see README for detailed explanation)

  ## Files Changed
  - `poker/agent.py`: Softmax implementation, expanded state encoding
  - `poker/play.py`: SARSA training loop, curriculum learning, model persistence
  - `poker/random_agent.py`: New baseline opponent class
  - `tests/test_agent.py`: Softmax unit tests
  - `README.md`: Results summary, future work, theoretical insights

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,4 +5,142 @@ sudo docker build ~/poker --tag=poker_docker
...
+  - **Root cause**: Not true self-play! Learning agent updates Q-function during episode, opponents frozen
+  - Opponents copy weights at episode start, then learning agent improves during episode
+  - Learning agent fights "slightly outdated versions of itself" → persistent edge
+- **Fix option A - Frozen evaluation**: Add separate eval episodes with no updates/exploration
+  - Measure win rate with frozen policies to check Nash convergence
+  - Current 65-68% is "training performance", not "equilibrium performance"
+- **Fix option B - Multi-agent RL**: All 3 agents update simultaneously
+  - More complex but would enable true Nash convergence
+  - Would require tracking separate Q-functions for each position
+
```
Comment (atorch) [inline README.md:291]: after self-play, make sure we still beat RandomAgent

Comment (atorch) [reply README.md:291]: related idea: make an `AugmentedRandomAgent` (think of a better name) that knows not to fold strong hands, or does so with very low probability -- make sure we beat that too

could be optional parameters for RandomAgent

Comment (atorch) [reply README.md:291]: `HeuristicAgent`
